instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java

```diff
@@ -61,98 +77,186 @@ public static Config get() {
   public abstract Map<String, String> getAllProperties();
 
   /**
-   * Returns a string property value or null if a property with name {@code name} did not exist.
-   *
-   * @see #getProperty(String, String)
+   * Returns a string-valued configuration property or {@code null} if a property with name {@code
+   * name} has not been configured.
    */
   @Nullable
   public String getProperty(String name) {
-    return getProperty(name, null);
+    return getRawProperty(name, null);
   }
 
   /**
-   * Retrieves a property value from the agent configuration consisting of system properties,
-   * environment variables and contents of the agent configuration file (see {@code
-   * ConfigInitializer} and {@code ConfigBuilder}).
-   *
-   * <p>In case any {@code get*Property()} method variant gets called for the same property more
-   * than once (e.g. each time an advice executes) it is suggested to cache the result instead of
-   * repeatedly calling {@link Config}. Agent configuration does not change during the runtime so
-   * retrieving the property once and storing its result in e.g. static final field allows JIT to do
-   * its magic and remove some code branches.
-   *
-   * @return A string property value or {@code defaultValue} if a property with name {@code name}
-   *     did not exist.
+   * Returns a string-valued configuration property or {@code defaultValue} if a property with name
+   * {@code name} has not been configured.
    */
   public String getProperty(String name, String defaultValue) {
-    return getAllProperties().getOrDefault(NamingConvention.DOT.normalize(name), defaultValue);
+    return getRawProperty(name, defaultValue);
   }
 
   /**
-   * Returns a boolean property value or {@code defaultValue} if a property with name {@code name}
-   * did not exist.
-   *
-   * @see #getProperty(String, String)
+   * Returns a boolean-valued configuration property or {@code null} if a property with name {@code
+   * name} has not been configured.
+   */
+  @Nullable
+  public Boolean getBoolean(String name) {
+    return getTypedProperty(name, Boolean::parseBoolean, null);
+  }
+
+  /**
+   * Returns a boolean-valued configuration property or {@code defaultValue} if a property with name
+   * {@code name} has not been configured.
    */
   public boolean getBooleanProperty(String name, boolean defaultValue) {
```

Comment:
I'm planning to rename all `get*Property()` methods to `get*()` in the next PR - those shorter names seem a bit nicer, and they're consistent with the SDK `ConfigProperties` interface too.

Reply:
is the plan to have this implement `ConfigProperties` then?

Reply:
Unfortunately that won't be possible with the current shape of this repo: `instrumentation-api` only uses OTel API, not the SDK, so we can't directly implement the `ConfigProperties` interface here.

Reply:
ok then. makes sense!